internal/controller/providers/pg/provider.go

```diff
@@ -109,6 +111,29 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 		status.Status = everestv1alpha1.AppStateRestoring
 	}
 
+	// Get PG pods to check if an upgrade is pending or in progress.
+	listOpts := &client.ListOptions{
+		Namespace:     p.DB.GetNamespace(),
+		LabelSelector: labels.SelectorFromSet(labels.Set{"app.kubernetes.io/component": "pg", "app.kubernetes.io/instance": p.DB.GetName()}),
+	}
+	podList := &corev1.PodList{}
+	if err := c.List(ctx, podList, listOpts); err != nil {
```

Comment:
One thing that concerns me - we will check the pods almost on each reconciliation iteration. Are there any other evidences of DB cluster upgrade process (maybe in pg-operator's DB cluster resource)?

Reply:
Unfortunately not, the PXC and PSMDB operators do have the version in their CRs but there's no such field in the PG CR :disappointed: I thought about opening a feature request for the cloud team to add a similar field.